InventoryManagementSystem.py

```diff
@@ -0,0 +1,198 @@
+            raise ValueError("Quantity can't be negative!")
+        self.name = name
+        self.price = price
+        self.quantity = quantity
+
+    def get_value(self) -> float:
+        return self.price * self.quantity
+
+    def __repr__(self):
+        return f'{self.name}, {self.price}, {self.quantity}'
+    
+    @staticmethod
+    def from_string(product_str: str):
+        name, price, quantity = product_str.strip().split(',')
+        return Product(name, float(price), int(quantity))
+    
+# BONUS Category class
+class Category:
+    def __init__(self, category: str):
+        self.category = category
+        pass
+
+# 2. Create an `InventoryManager` class that will manage a list of `Product` objects. It should have the following methods:
+#   -[x] `add_product(self, product)`: Adds a new product to the inventory
+#   -[x] `remove_product(self, product_name)`: Removes a product from the inventory by name
+#   -[x] `update_quantity(self, product_name, new_quantity)`: Updates the quantity of a product
+#   -[x] `get_total_inventory_value(self)`: Calculates and returns the total value of all products in the inventory
+#   -[x] `list_low_stock_products(self, threshold)`: Returns a list of products with quantity below the given threshold
+
+class InventoryManager:
+    def __init__(self, filename="inventory.txt"):
+        self.filename = filename
+        self.Inventory = self.load_inventory()
+        #self.Inventory = [] # Empty list to hold Inventory
+
+    def load_inventory(self):
+        inventory = []
+        try:
+            with open(self.filename, 'r') as file:
+                for line in file:
+                    inventory.append(Product.from_string(line))
+        except FileNotFoundError:
+            print(f"{self.filename} not found.")
+        return inventory
+    
+    def save_inventory(self):
+        with open(self.filename, "w") as file:
+            for product in self.Inventory:
+                file.write(f'{product}\n')
+    
+    def add_product(self, product):
+        for existing_product in self.Inventory:
+            if product.name == existing_product.name:
+                print(f"{product.name} already exists. Please enter a new product, or select '3' to update quantity.")
+                return
+        self.Inventory.append(product)
+        self.save_inventory()
+        print(f"Product: {product.name} (x{product.quantity}) has been added at ${product.price}")
+        return
+                
+
+    # -[ ] Look into using filter() or list comprehension!!
+    def remove_product(self, product_name):
+        for product in self.Inventory:
+            if product.name == product_name:
+                self.Inventory.remove(product)
+                self.save_inventory()
+                print(f"Removed {product_name}")
+                return
+        print(f"Product {product_name} not found in inventory.")
+
+    def update_quantity(self, product_name, new_quantity):
```

Comment:
add method strings please:

```python
def update_quantity(self, product_name: str, new_quantity: int) -> None:
    """Updates the quantity of a product in inventory.
    
    Args:
        product_name: Name of product to update
        new_quantity: New quantity to set
        
    Raises:
        ValueError: If new_quantity is negative
    """
```

Reply:
Please add to each method

Reply:
Docstrings have been added to all methods